text/0000-item-grouping.md

```diff
@@ -0,0 +1,152 @@
+- Feature Name: `item_grouping`
+- Start Date: 2015-02-13
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Allow an anonymous `impl { ... }` for grouping items without introducing a module.
+
+# Motivation
+
+Many libraries today contain code like
+
+```rust
+#[cfg(target_os = "windows")]
+mod bar;
+
+#[cfg(target_os = "windows")]
+pub use bar::Baz;
+
+#[cfg(target_os = "windows")]
+fn quux() { ... }
+```
+
+With the new syntax, we can eliminate the duplication of `cfg` directives:
+
+```rust
+#[cfg(target_os = "windows")]
+impl {
```

Comment:
Sometimes it can, but sometimes you have a different set of items available on each platform.  In that case you need a glob `pub use` or some `cfg`'d `pub use`s.